blood/src/main/java/cz/cuni/mff/d3s/blood/node_type_tracker/NodeTypeTrackerCollector.java

```diff
@@ -0,0 +1,78 @@
+package cz.cuni.mff.d3s.blood.node_type_tracker;
+
+import cz.cuni.mff.d3s.blood.report.Report;
+import cz.cuni.mff.d3s.blood.report.dump.ManualDump;
+import org.graalvm.compiler.graph.Node;
+import org.graalvm.compiler.nodes.StructuredGraph;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class NodeTypeTrackerCollector {
```

Comment:
Copy-pasted from `DependencyMatrixCollector`. Maybe extract the matrix code to a `Matrix<ValueType>` class?

Reply:
This won't be the case soon. `DependencyMatrixCollector`'s matrix will change quite a bit in a short amount of time due to planned changes. Probably won't fix.